src/main/pageobjects/amz.productDetails.page.ts

```diff
@@ -1,92 +1,109 @@
-import { ChainablePromiseElement } from 'webdriverio';
-import Page from './page';
-import * as config from 'config'
-
-let pdtsearchString : string = config.get('ProductData.searchProduct')
-let pdtexpectedString : string = config.get('ProductData.expectedString')
-
-class ProductDetailsPage extends Page {
-
-    
-    public get pdtToSearch() {
-        return <any>$('(//*[contains(text(), "'+pdtexpectedString+'")])//parent::a/..');
-    }
-
-    public get searchField() {
-        return <any>$('#twotabsearchtextbox');
-    }
-    public get btnSearch() {
-        return <any>$('#nav-search-submit-button');
-    }
-
-    public get searchResults() {
-        return <any>$$('//div/h2/a/span[contains(text(),"'+pdtexpectedString+'")]');
-    }
-
-    public get emiOptions(){
-        return <any>$('//*[contains(text(),"EMI options")]');
-    }
-
-
-    public get pdtImage() {
-        return <any>$('//img[contains(@alt,"'+pdtexpectedString+'") and contains(@onload,"markFeatureRenderForImageBlock()")]');
+import { ChainablePromiseElement } from "webdriverio";
+import Page from "./page";
+import * as config from "config";
+
+let pdtsearchString: string = config.get("ProductData.searchProduct");
+let pdtexpectedString: string = config.get("ProductData.expectedString");
+
+class productDetailsPage extends Page {
+  public get productToSearch() {
+    return <any>(
+      $('(//*[contains(text(), "' + pdtexpectedString + '")])//parent::a/..')
+    );
+  }
+
+  public get searchField() {
+    return <any>$("#twotabsearchtextbox");
+  }
+  public get btnSearch() {
+    return <any>$("#nav-search-submit-button");
+  }
+
+  public get searchResults() {
+    return <any>(
+      $$('//div/h2/a/span[contains(text(),"' + pdtexpectedString + '")]')
+    );
+  }
+
+  public get emiOptions() {
+    return <any>$('//*[contains(text(),"EMI options")]');
+  }
+
+  public get productImage() {
+    return <any>(
+      $(
+        '//img[contains(@alt,"' +
+          pdtexpectedString +
+          '") and contains(@onload,"markFeatureRenderForImageBlock()")]'
+      )
+    );
+  }
+
+  public get fullScreenImage() {
+    return <any>$('//img[@class="fullscreen"]');
+  }
+
+  public get ZoomImageTitle() {
+    return <any>$('//div[@id="ivTitle"]');
+  }
+
+  public async searchAProduct(searchProduct) {
+    await this.searchField.setValue(searchProduct);
+    await this.btnSearch.clickWhenDisplayed();
+  }
+
+  public async selectProductFromResults() {
+    await this.productToSearch.waitTillInterval(3000);
+    await (await this.productToSearch).click();
+  }
```

Comment:
instead of waiting and then clicking, can you use the custom command here?

Reply:
Updated the clickWhenDisplayed custom command is updated